File: chall01/krusthol.py
#!/usr/bin/python3
import sys
import string

def bad_character(str):
    good = set(string.ascii_lowercase + ' ')
    for c in str:
        if c not in good:
            return True
    return False

if len(sys.argv) != 2:
    sys.exit("usage: ./krusthol.py <a-zA-Z string>")
arg = sys.argv[1].lower()
if len(arg) == 0 or bad_character(arg):
    sys.exit("usage: ./krusthol.py <a-zA-Z string>")
for c in arg:
    if c == 'a':
        sys.stdout.write('.-')
    elif c == 'b':
        sys.stdout.write('-...')
    elif c == 'c':
        sys.stdout.write('-.-.')
    elif c == 'd':
        sys.stdout.write('-..')
    elif c == 'e':
        sys.stdout.write('.')
    elif c == 'f':
        sys.stdout.write('..-.')
    elif c == 'g':
        sys.stdout.write('--.')
    elif c == 'h':
        sys.stdout.write('....')
    elif c == 'i':
        sys.stdout.write('..')
    elif c == 'j':
        sys.stdout.write('.---')
    elif c == 'k':
        sys.stdout.write('-.-')
    elif c == 'l':
        sys.stdout.write('.-..')
    elif c == 'm':
        sys.stdout.write('--')
    elif c == 'n':
        sys.stdout.write('-.')
    elif c == 'o':
        sys.stdout.write('---')
    elif c == 'p':
        sys.stdout.write('.--.')
    elif c == 'q':
        sys.stdout.write('--.-')
    elif c == 'r':
        sys.stdout.write('.-.')
    elif c == 's':
        sys.stdout.write('...')
    elif c == 't':
        sys.stdout.write('-')
    elif c == 'u':
        sys.stdout.write('..-')
    elif c == 'v':
        sys.stdout.write('...-')
    elif c == 'w':
        sys.stdout.write('.--')
    elif c == 'x':
        sys.stdout.write('-..-')
    elif c == 'y':
        sys.stdout.write('-.--')
    elif c == 'z':
        sys.stdout.write('--..')
    elif c == ' ':
        sys.stdout.write(' ')
print("")
sys.exit(0)

File: chall05/krusthol.rb
#!/usr/bin/env ruby
#ruby 2.5.1p57
require 'open-uri'

counter = 0
checked_pages = []
if ARGV[0] != nil
    starting_point = 'https://en.wikipedia.org/wiki/' + ARGV[0]
else
    puts "Road to Philosophy script usage: ./krusthol.rb [Wikipedia article]"
    exit
end
begin
contents = open(starting_point).read
rescue
        puts "no wikipedia page for this word!"
        exit
end
split_p = contents.split('<p>')
i = 1
road = split_p[i].scan(/(?<=<a href="\/wiki\/).[^"]*/)
while road[0] == nil do
    i = i + 1
    road = split_p[i].scan(/(?<=<a href="\/wiki\/).[^"]*/)    
end
checked_pages.push(road[0])
counter = counter + 1
puts ("Going to " + road[0] + " (counter: #{counter})")
while !checked_pages.include?('Philosophy') do
    contents = open('https://en.wikipedia.org/wiki/' + checked_pages.last).read
    split_p = contents.split('<p>')
    i = 1
    road = nil
    while road == nil do
        road = split_p[i].scan(/(?<=<a href="\/wiki\/).[^"]*/)
        i = i + 1
    end
    i = 0
    while checked_pages.include?(road[i]) do
        puts ("Skip " + road[i] + ", already seen this one")
        i = i + 1    
    end
    checked_pages.push(road[i])
    counter = counter + 1
    puts ("Going to " + road[0] + " (counter: #{counter})")
end
puts ("!!! Reach Philosophy !!!")
